experimental/GITFans/GITFans.jl

```diff
@@ -416,19 +438,14 @@ function fan_traversal(orbit_list, q_cone, perm_actions)
 
         # note that we run also over elements added inside the loop
         current_cone_list = cones_from_bitlist(orbit_list, current_hash)
-        intersected_cone = Polymake.polytope.intersection(current_cone_list...)
-        facets = intersected_cone.FACETS
-        facets = convert(Matrix{Rational{BigInt}},facets)
-        facet_points = Vector{Rational{BigInt}}[
-                           Polymake.polytope.facet(intersected_cone, i-1).REL_INT_POINT
-                           for i in 1:size(facets, 1)]
+        intersected_cone = intersect(current_cone_list)
+        ic_facets = -linear_inequality_matrix(facets(intersected_cone))
+        facet_points = [Vector{T}([relative_interior_point(f)...]) for f in faces(intersected_cone, dim(intersected_cone)-1)]
 
         neighbor_hashes = []
         for i in 1:length(facet_points)
-            if any(x->x==0, q_cone_facets_converted*facet_points[i])
-                continue
-            end
-            push!(neighbor_hashes, get_neighbor_hash(orbit_list, facet_points[i], facets[i, :]))
+          !any(f->contains(f, facet_points[i]), faces(q_cone, dim(q_cone)-1)) || continue
```

Comment:
Could also write it like this:
```suggestion
          !any(contains(facet_points[i]), faces(q_cone, dim(q_cone)-1)) || continue
```
and if #2142 is resolved then could even write
```suggestion
          !any(contains(facet_points[i]), facets(Cone, q_cone)) || continue
```

Reply:
This requires more work than suggested in #2142:
```
julia> c = positive_hull([1 0; 0 1])
Polyhedral cone in ambient dimension 2

julia> fc = facets(Cone{QQFieldElem}, c)
2-element SubObjectIterator{Cone{QQFieldElem}}:
 Polyhedral cone in ambient dimension 2
 Polyhedral cone in ambient dimension 2

julia> [rays_modulo_lineality(f) for f in fc]
2-element Vector{NamedTuple{(:rays_modulo_lineality, :lineality_basis), Tuple{SubObjectIterator{RayVector{QQFieldElem}}, SubObjectIterator{RayVector{QQFieldElem}}}}}:
 (rays_modulo_lineality = [[1, 0]], lineality_basis = [[0, 1]])
 (rays_modulo_lineality = [[0, 1]], lineality_basis = [[1, 0]])
```
As you can see, these "facets" are just the linear halfspaces as a different type. I was just as confused as you and happy that I found a workaround for now. I am currently discussing with @benlorenz what to do.

Reply:
Thank you for looking into this :-)

Reply:
OK, I see:
```julia-repl
julia> q_cone = cone_from_inequalities([-1 0 0; 0 -1 0; 0 0 -1])
Polyhedral cone in ambient dimension 3

julia> f1 = faces(q_cone, dim(q_cone)-1);

julia> f2 = facets(Cone{QQFieldElem}, q_cone);

julia> [rays_modulo_lineality(f) for f in f1]
3-element Vector{NamedTuple{(:rays_modulo_lineality, :lineality_basis), Tuple{SubObjectIterator{RayVector{QQFieldElem}}, SubObjectIterator{RayVector{QQFieldElem}}}}}:
 (rays_modulo_lineality = [[0, 1, 0], [0, 0, 1]], lineality_basis = [])
 (rays_modulo_lineality = [[1, 0, 0], [0, 0, 1]], lineality_basis = [])
 (rays_modulo_lineality = [[1, 0, 0], [0, 1, 0]], lineality_basis = [])

julia> [rays_modulo_lineality(f) for f in f2]
3-element Vector{NamedTuple{(:rays_modulo_lineality, :lineality_basis), Tuple{SubObjectIterator{RayVector{QQFieldElem}}, SubObjectIterator{RayVector{QQFieldElem}}}}}:
 (rays_modulo_lineality = [[1, 0, 0]], lineality_basis = [[0, 1, 0], [0, 0, 1]])
 (rays_modulo_lineality = [[0, 1, 0]], lineality_basis = [[1, 0, 0], [0, 0, 1]])
 (rays_modulo_lineality = [[0, 0, 1]], lineality_basis = [[1, 0, 0], [0, 1, 0]])
```

So yeah, these are different. Thanks also for the explanation on gather.